### ISSUE TITLE
Options defined in extended InfinityModel overwritten by infinity service defaults

### ISSUE THREAD

**Issue by callovarne** [ember-infinity/ember-infinity#336]
## Background

In _ember-infinity_ pre 1.0, options like _perPageParam_ could be set up app-wide in the previously recommended route mixin. With 1.0's API changes, that appears to no longer be possible.

Any options given in a subclass of _InfinityModel_ are effectively ignored. Instead, such options must be passed in with _every_ invocation of _this.infinity.model()_, causing code duplication in the implementation and added maintenance cost.

## Steps
1. Extend _InfinityModel_:

    ```js
    import InfinityModel from 'ember-infinity/lib/infinity-model';

    export default InfinityModel.extend({
      perPageParam: 'limit',
      totalPagesParam: 'meta.totalPages'
    });
    ```

1. Provide the extended _InfinityModel_ when invoking the _infinity_ service's _model_ method:

    ```js
    import InfinityModel from 'client/models/infinity-model';
    …
      infinity: service(),
      model() {
        return this.infinity.model('post', {}, InfinityModel);
      }
    ```

Expected: request is sent with `limit=100`.
Actual: request is sent with `per_page=100`.

## Notes

The problem appears to be in the way the _infinity_ service handles options. For example, _perPageParam_ has a value of `per_page` in [addon/lib/infinity-model.js](https://github.com/ember-infinity/ember-infinity/blob/master/addon/lib/infinity-model.js#L114), but is reassigned in [addon/services/infinity.js](https://github.com/ember-infinity/ember-infinity/blob/master/addon/services/infinity.js#L199) to `per_page` again.

A workaround for options affecting the request params might be to hook into _buildParams_, but the same is not possible for options affecting how the response is interpreted (e.g. _totalPagesParam_).

I suspect the way forward involves updating _paramsCheck_ in [addon/utils.js](https://github.com/ember-infinity/ember-infinity/blob/master/addon/utils.js#L36) to account for options given in the extended _InfinityModel_. That might look like:

```diff
--- paramsCheck(optionParam, defaultParam) {
+++ paramsCheck(key, options, extendedInfinityModel) {
```

...where instead of passing in values, a key and the objects to check are given. If a _options_ has a value for the given _key_, use it. Otherwise, use the default defined in _InfinityModel_.

What do you think about this approach?

**Comment by johncalvinyoung** [comment]
Observed this bug last week, glad to hear you might have a way forward. Hadn’t actually sorted it out yet as I just needed to override buildParams anyway.

**Comment by snewcomer** [comment]
#337 has been merged! 1.4.6.  Thank you!
